src/Chainweb/Chainweb/MinerResources.hs

```diff
@@ -101,11 +105,14 @@ runMiner v mr = do
         pure $ localTest gen miners
 
     powMiner :: IO (BlockHeader -> IO BlockHeader)
-    powMiner = case _configRemoteMiners conf of
-        [] -> pure $ localPOW v
-        rs -> do
+    powMiner = case g $ _configRemoteMiners conf of
+        Nothing -> pure $ localPOW v
+        Just rs -> do
             m <- newManager defaultManagerSettings
-            pure . remoteMining m $ map f rs
+            pure $ remoteMining m rs
+
+    g :: Set HostAddress -> Maybe (NonEmpty BaseUrl)
+    g = fmap (NEL.map f) . NEL.nonEmpty . S.toList
```

Comment:
What is the benefit of this? I think I found the code before more readable. Is the new code safer?

Reply:
Before this PR, the logic in the `remoteMiner` function in `Miners.hs` was incorrect. `NonEmpty` + `traverseConcurrently` + `These` lets me solve it elegantly.